### PR TITLE
Upload a digest .docx file, check an email is sent

### PR DESCRIPTION
**Pull request by giorgiosironi**
For https://github.com/elifesciences/jira-import/issues/4325

The scope of end2end is to test that the full workflow has run and that an email can be sent, while details on the email's content should  be tested at the project level for performance and ease of reproducibility.

The test:

- generates a random article id and a related digest `.docx` document using it in its DOI 
- uploads it to the `end2end-elife-bot-digests-input` bucket
- checks that there is an email caught, with the expected subject

### CODE REVIEW

File: spectrum/checks.py [hunk 1]
```diff
@@ -743,43 +741,7 @@ def _is_present(self, url_page_template, id):
             page = page + 1
 
 def _poll(action_fn, error_message, *error_message_args):
```
Comment (giorgiosironi) [inline spectrum/checks.py]: moved into `spectrum.polling` to try to cut down on the size of this module

File: spectrum/generator.py [hunk 2]
```diff
@@ -83,6 +84,22 @@ def _substitute_article_id(text):
...
 
+def digest_doc(template_id):
+    standard_input = 'spectrum/templates/DIGEST 99999.docx'
+    article_id = generate_article_id(template_id)
+    target_filename = '%s/DIGEST %s.docx' % (COMMON['tmp'], article_id)
+
+    word_document = docx.Document(standard_input)
+    doi_paragraphs = [p for p in word_document.paragraphs if p.runs[0].text == 'FULL ARTICLE DOI\n']
+    assert len(doi_paragraphs) == 1, "Wrong number of FULL ARTICLE DOI paragraphs: %s" % [p.text for p in word_document.paragraphs]
+    doi_paragraphs[0].runs[1].text = 'https://doi.org/10.7554/eLife.%s' % article_id
```
Comment (gnott) [inline spectrum/generator.py]: You make it look so simple!

File: spectrum/test_digest.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+from spectrum import input
+
+SIMPLEST_ARTICLE_ID = 15893
+
+@pytest.mark.bot
+#@pytest.mark.digests
+def test_digest_lifecycle(generate_digest):
+    digest = generate_digest(SIMPLEST_ARTICLE_ID)
+    input.DIGESTS_BUCKET.upload(digest.filename(), id=digest.article_id())
+    checks.BOT_EMAILS.wait_email(subject='Digest: Anonymous_%s' % digest.article_id())
```
Comment (gnott) [inline spectrum/test_digest.py]: This subject will match if it was a success :+1:

**Comment by giorgiosironi** [comment]
This uses a .docx rather than a full .zip, but it's a good start. Will extend later by using a .zip containing the image too.

**Pull request merged**